flasher.cpp

```diff
@@ -331,13 +293,8 @@ FlashingInfo Flasher::Flash()
     const char *data_firmware = byte_array.data() + kSignatureSize;
 
     if (serial_port_.isOpen()) {
-        flashing_info.success = SendKey();
-        if (!flashing_info.success) {
-            flashing_info.title = "Error";
-            flashing_info.description = "Send key problem";
-        }
-    }
-    else {
+        flashing_info.success = true;
```

Comment:
Instead of setting success flag to true, we can start here with signature check.

```suggestion
    if (serial_port_.isOpen()) {
        flashing_info.success = SendMessage(kCheckSignatureCmd, sizeof(kCheckSignatureCmd), kSerialTimeoutInMs);

        if (!flashing_info.success) {
            flashing_info.title = "Flashing process failed";
            flashing_info.description = "Check signature problem";
        }
    }
    else {
        flashing_info.title = "Error";
        flashing_info.description = "Serial port is not opened";
    }
```